### ISSUE TITLE
Updating start_date with REST API

### ISSUE THREAD

**Issue by usenetCoder** [paypal/PayPal-REST-API-issues#185]
I see in the documentation (https://developer.paypal.com/docs/api/payments.billing-agreements#definition-patch) that you patch the start_date for a billing agreement. This would be useful when offering a trial with limited functionality and if the user wants to gain full functionality before the trial expires you should be able to set the start date to a date/time sooner then the initial trial terms. 

But when looking over the PHP REST SDK I don't see any option to update the start_date. Is the possible with PHP? What would be the best way to offer a trial and start the agreement early if the user wants full functionality sooner?

**Comment by usenetCoder** [comment]
Is there a better place to ask this question? As it does appear to be an issue with the API. If so where should I ask it?

**Comment by pp-randy** [comment]
The PHP SDK supports patching billing agreements https://github.com/paypal/PayPal-PHP-SDK/blob/master/sample/billing/UpdateBillingAgreement.php.  

Have you tried updating the sample script for start_date?

**Comment by usenetCoder** [comment]
Yes. First we create the agreement, then patch with new address, description, and start date and nothing takes effect. Ideas? Does it work for you?

**Comment by usenetCoder** [comment]
This is how it looks when I'm padding in the start_date. The description and shipping address will update but not the start_date. Really need this guys. Please help. Also I can't use the Classic API as it shows it's deprecated as of January 1, 2017.

![update](https://user-images.githubusercontent.com/27294280/35015987-60d06878-fadc-11e7-88a5-6d4866a2fbdb.JPG)

**Comment by pp-randy** [comment]
I've tested the same and the `start_date` is not being modified either by the REST API or Classic API. 

I will escalate internally and keep this issue open for tracking.

**Comment by usenetCoder** [comment]
Thanks for testing this Randy. We offer a SaaS poduct which has a free trial. With the free trial the product is limited. The trial lasts for 14 days. If the customer decides that they want full functionality early we currently can't do that with PayPal. We have to tell the customer to cancel their current billing agreement and signup for a new one which is very inconvenient for our customers, sometimes they forget to cancel their trial agreement and are double billed which makes us look bad.

As SaaS products before more and more popular I can only assume that offering trials will too. The ability to modify the start_date should be fairly important I would think. Please keep us updated on this as we can't move forward with development until this solution is available. Thank You.

**Comment by pp-randy** [comment]
I will also submit a feature request for an easy way to end a trial period, I will update with a new issue to track that as well. I understand they are related in your use case, but `start_date` update is failing even without a trial.

**Comment by usenetCoder** [comment]
Thank you very much Randy. Your help is greatly appreciated.

**Comment by pp-randy** [comment]
https://github.com/paypal/PayPal-REST-API-issues/issues/187

**Comment by pp-randy** [comment]
Start_Date changing is only supported for credit card funded subscriptions.  I am ensuring documentation is updated to call this out.  I am additionally following up to see if this can be added for PayPal funded subscriptions.

**Comment by usenetCoder** [comment]
Thank you very much for the quick update. If the start date update could be added to PayPal funding as well that would be great. Please keep us updated on the progress of adding that functionality.

**Comment by usenetCoder** [comment]
Hey @pp-randy any updates on this?

**Comment by pp-randy** [comment]
The only existing way is to create a new billing agreement unfortunately.

**Comment by usenetCoder** [comment]
>Start_Date changing is only supported for credit card funded subscriptions. I am ensuring documentation is updated to call this out. I am additionally following up to see if this can be added for PayPal funded subscriptions.

Any updates if this can / will be added for PayPal funding?

**Comment by pp-randy** [comment]
I will follow up; unfortunately, I do not have an update at this time.

**Comment by ipula** [comment]
I have same issue... need to update start date functionality @randstraw any updates on this?
